heft-plugins/heft-serverless-stack-plugin/src/ServerlessStackPlugin.ts

```diff
@@ -0,0 +1,229 @@
+  ScopedLogger
+} from '@rushstack/heft';
+// TODO: Expose this API
+import { SubprocessTerminator } from '@rushstack/heft/lib/utilities/subprocess/SubprocessTerminator';
+import { FileSystem, Import } from '@rushstack/node-core-library';
+
+const PLUGIN_NAME: string = 'ServerlessStackPlugin';
+const TASK_NAME: string = 'heft-serverless-stack';
+const SST_CLI_PACKAGE_NAME: string = '@serverless-stack/cli';
+
+/**
+ * Options for `ServerlessStackPlugin`.
+ *
+ * @public
+ */
+export interface IServerlessStackPluginOptions {}
+
+/** @public */
+export class ServerlessStackPlugin implements IHeftPlugin<IServerlessStackPluginOptions> {
+  public readonly pluginName: string = PLUGIN_NAME;
+
+  private _logger!: ScopedLogger;
+
+  public apply(
+    heftSession: HeftSession,
+    heftConfiguration: HeftConfiguration,
+    options: IServerlessStackPluginOptions
+  ): void {
+    this._logger = heftSession.requestScopedLogger(TASK_NAME);
+
+    const sstParameter: IHeftFlagParameter = heftSession.commandLine.registerFlagParameter({
+      associatedActionNames: ['test', 'build', 'start'],
+      parameterLongName: '--sst',
+      // Once https://github.com/serverless-stack/serverless-stack/issues/1537 is fixed, we may be
+      // eliminate the need for this parameter.
+      description: 'Invokes the SST postprocessing - requires AWS credentials'
+    });
+
+    const sstStageParameter: IHeftStringParameter = heftSession.commandLine.registerStringParameter({
+      associatedActionNames: ['test', 'build', 'start'],
+      parameterLongName: '--sst-stage',
+      argumentName: 'STAGE_NAME',
+      description:
+        'Specifies the Serverless Stack stage; equivalent to to the "--stage" parameter from the "sst" CLI'
+    });
+
+    let sstCliPackagePath: string;
+
+    try {
+      sstCliPackagePath = Import.resolvePackage({
+        packageName: SST_CLI_PACKAGE_NAME,
+        baseFolderPath: heftConfiguration.buildFolder
+      });
+    } catch (e) {
+      throw new Error(
+        `The ${TASK_NAME} task cannot start because your project does not seem to have a dependency on the` +
+          ` "${SST_CLI_PACKAGE_NAME}" package: ` +
+          e.message
+      );
+    }
+
+    const sstCliEntryPoint: string = path.join(sstCliPackagePath, 'bin/scripts.js');
+    if (!FileSystem.exists(sstCliEntryPoint)) {
+      throw new Error(
+        `The ${TASK_NAME} task cannot start because the entry point was not found:\n` + sstCliEntryPoint
+      );
+    }
+
+    this._logger.terminal.writeVerboseLine('Found SST package in' + sstCliPackagePath);
+
+    heftSession.hooks.build.tap(PLUGIN_NAME, (build: IBuildStageContext) => {
+      build.hooks.bundle.tap(PLUGIN_NAME, (bundle: IBundleSubstage) => {
+        bundle.hooks.configureWebpack.tap(
+          { name: PLUGIN_NAME, stage: Number.MAX_SAFE_INTEGER },
+          (webpackConfiguration: unknown) => {
+            // Discard Webpack's configuration to prevent Webpack from running
+            return null;
```

Comment:
This should probably only happen with the `--sst` parameter

Reply:
fixed